### DISCUSSION TITLE
Columns grouping

### DISCUSSION THREAD

**Discussion by churikdy** [hnolCol/instantclue#73] [Q&A]
Does anybody know how to specify columns grouping when uploading data to InstantClue?

**Comment by hnolCol** [comment]
Hi, 

you can select "Annotate Groups" from the context menu, please note that only Numeric Floats can be grouped. If your data are of type Integer you will have to change it (Column operation. .. -> Change datatype) before. 

![image](https://github.com/hnolCol/instantclue/assets/24191910/968a7cca-780f-4b5d-92ae-0a77f089e6d0)

A dialog should open and look like this. You can then use drag & drop to group the columns. Don't forget to Name the Groups (individual columns in a group) as well as the overall grouping (example: MyFirstGrouping). 

![image](https://github.com/hnolCol/instantclue/assets/24191910/ce1b1039-2451-4435-a48a-fc38933fd710)

You can read more about groupings [in the WIKI](https://github.com/hnolCol/instantclue/wiki/Groupings) 
Most importantly, if you have many columns to be grouped, it is advisable to use the **Group by split string** functionality which is explained in the WIKI and available by right-clicking in the Grouping-Dialog. Please let me know if you have still some open questions. 
Noteworthy, you can save the grouping to a json file as shown here. 

![image](https://github.com/hnolCol/instantclue/assets/24191910/92da926c-72f3-489f-b27c-716b9d86c2bf)

You can then load the grouping. Please note that it recognizes the names only, meaning if you change the name it will not be recognized from a saved grouping. The groups are indicated by the colored dots next to the column names. 

Once the grouping is assigned you can perform ANOVA and pairwise comparisons from the Groupings context menu. In addition, for hierarchical clusterings, you will be asked if you want to show the groupings in the heatmap. 

Again, please let me know if you are missing some functionality.

**Comment by churikdy** [comment]
Hi,
Thank you so much for your help! It works beautifully!
At the moment everything is clear and sorry that I couldn't find it myself,
now it seems obvious. Way more to learn.
Thank you again,
Dmitry

**Reply by hnolCol** [reply]
Superb, happy to hear that it works as expected.
Just one additional comment, when you are plotting a hierarchical clustering, you can add the legend for the groupings using the context menu as shown below.

![image](https://github.com/hnolCol/instantclue/assets/24191910/3e561306-b529-42f7-a217-b43fb45fff86)

**Comment by churikdy** [comment]
Thank you, that is also a very useful feature!
